### DISCUSSION TITLE
How to configure MQTT channel

### DISCUSSION THREAD

**Discussion by aliraza995** [ibm-messaging/mq-container#497] [General]
I am working on a project where we need to use mqtt protocol to communicate with IBM MQ. I was trying to configure ibm mq  docker container with mqtt channel configured but unable to configure it so far. Can you suggest if its possible if yes. Can you suggest how?

**Comment by callumpjackson** [comment]
You will need to rebuild the container image and enable the MQTT feature. This is down by editing the [install-mq.sh](https://github.com/ibm-messaging/mq-container/blob/master/install-mq.sh) file from:
`export genmqpkg_incmqxr=0`
to
`export genmqpkg_incmqxr=1`
Then as part of the container build process the MQTT feature should be included.

**Comment by arthurbarr** [comment]
The sample container doesn't have MQTT installed or configured, so you'll need to build your own image.  I'm not aware of anyone who has done it, but it should certainly be possible.  A few pointers:

1. You'll need to install the MQXR package, which you can enable in the [install-mq.sh](https://github.com/ibm-messaging/mq-container/blob/d7c7b52492c6b1ae600bf3b8ccd9724ffd45e361/install-mq.sh#L52) script.
2. I don't think the XR server gets started automatically, so you'll need to start the XR server at runtime.
   * You could do it through an MQ "service" configured via MQSC
   * We already start the MQ web server in [post_init.go](https://github.com/ibm-messaging/mq-container/blob/master/cmd/runmqserver/post_init.go), so you could do it from the Go code there.
   * Alternatively, I _think_ you could create a separate container which shares an IPC namespace (as Pods do in Kubernetes), and run it in a separate container.
4. If you decide to run it in the same container, you might want to mirror the logs to the container stdout, which we already do for the [other logs](https://github.com/ibm-messaging/mq-container/blob/d7c7b52492c6b1ae600bf3b8ccd9724ffd45e361/cmd/runmqserver/main.go#L207).
